### PR TITLE
Update dev doc 

### LINKED ISSUE #1166: Update developer-guide.md with new instructions of outerloop testing

### PR DESCRIPTION
**Pull request by zhenlan**
Add new instructions of outerloop testing
Also updated dev doc to include build.sh for x-plat.

fixes #1166 

skip ci please

### CODE REVIEW

File: Documentation/developer-guide.md [hunk 1]
```diff
@@ -23,58 +23,61 @@ Please read [Contributing](https://github.com/dotnet/wcf/blob/master/Documentati
...
+
+To run the scenario tests on Windows, execute this CMD from the root of the repository. It will automatically start a
+self-hosted WCF test service on the same Windows machine and shut it down at the end of execution of all tests.
 
 ```
     build.cmd /p:WithCategories=OuterLoop
 ```
+To run scenario tests against an already running WCF test service, you will need to specify the base address 
+of the WCF test service by using the `/p:ServiceUri` parameter. This is also necessary when running
+scenrio tests on Linux or OS X as the WCF service can only run on Windows.
```
Comment (roncain) [inline Documentation/developer-guide.md]: Nit: "scenario" spelling

File: Documentation/developer-guide.md [hunk 2]
```diff
@@ -23,58 +23,61 @@ Please read [Contributing](https://github.com/dotnet/wcf/blob/master/Documentati
...
+```
+or run this command to run scenario tests on Linux or OS X.
+Note some of security related tests require extra manual setup to work, 
+see [Linux manual test guide](https://github.com/dotnet/wcf/blob/master/Documentation/manualtest-guide.md) for details.
+```
+    ./build.sh /p:WithCategories=OuterLoop /p:ServiceUri=[WCFTestServiceBaseAddress]
+```
+
+You will need to replace `[WCFTestServiceBaseAddress]` in above commands with an actual value.
+See [Scenario test guide](https://github.com/dotnet/wcf/blob/master/Documentation/scenario-test-guide.md)
```
Comment (roncain) [inline Documentation/developer-guide.md]: Recommend a couple examples so they know [WcfTestServiceBaseAddress] is just the machine name (not some service endpoint), such as:
```
build.cmd /p:WithCategories=OuterLoop /p:ServiceUri=localhost
    or
./build.sh /p:WithCategories=OuterLoop /p:ServiceUri=MyWindowsServer
 
```

File: Documentation/scenario-test-guide.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Copy script `src\System.Private.ServiceModel\tools\scripts\SetupWcfIISHostedService.cmd` to the Windows machine and run it in an elevated command window.
+```
+    SetupWcfIISHostedService.cmd {id}
+```
+
+where `id` is an identifier that will be appended to the name of the IIS application to be created, for example, command `SetupWcfIISHostedService.cmd Test` will create an IIS aplication named "WCFService*Test*". This script will create a clone of WCF GitHub repository, an IIS application pool and an IIS application from `src\System.Private.ServiceModel\tools\IISHostedWcfService`. The script will also install necessary test certificates and add IIS HTTPS binding for testing.
+
+This is one time setup. Once you set it up, you do not need to build or start the service manually. IIS will build and activate the service automatically upon the first request from tests. 
+
+Runing WCF Scenario Tests
```
Comment (roncain) [inline Documentation/scenario-test-guide.md]: Nit: "Running" spelling

File: Documentation/developer-guide.md [hunk 4]
```diff
@@ -23,58 +23,61 @@ Please read [Contributing](https://github.com/dotnet/wcf/blob/master/Documentati
...
 
 Obtaining code coverage
 =======================
 You can also obtain detailed code coverage information by including an additional property
 when you run the build script.  For example, this CMD runs the scenario tests and collects
-code coverage numbers:
+code coverage numbers on Windows:
 ```
     build.cmd /p:WithCategories=OuterLoop /p:Coverage=true
 ```
```
Comment (roncain) [inline Documentation/developer-guide.md]: I recommend we tell them that after this is run, they can find the results under bin\tests\coverage\index.html.  Someone needs to verify this -- it's just where I think the results go now.

File: Documentation/scenario-test-guide.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+Scenario Testing Guide
+======================
+
+Overview
+--------
+WCF scenario tests are also referred to as outerloop tests. These are tests involving communication between WCF client tests and WCF test services. The WCF client tests running on .NET Core support cross platform, which means they can run on all supported platforms such as Windows, various Linux distros, Mac OS X as well as Universal Windows Platform (UWP). The WCF test services are built on full .NET Framwork that is supported on Windows only.
+
+Regardless which platform where you run WCF client tests, you will need a WCF test service for scenario tests to work. WCF Test services can be hosted in two ways like WCF users would usually do, ie., self-hosted and IIS-hosted. The following sections provide instructions of how to host WCF test services and how to run WCF scenario tests.
```
Comment (roncain) [inline Documentation/scenario-test-guide.md]: Recommend ending the sentence "..., self-hosted or IIS-hosted on a Windows machine using the full .NET Framework (for example Windows Server 2012 R2 or Windows10 Enterprise)."

File: Documentation/scenario-test-guide.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    SetupWcfIISHostedService.cmd {id}
+```
+
+where `id` is an identifier that will be appended to the name of the IIS application to be created, for example, command `SetupWcfIISHostedService.cmd Test` will create an IIS aplication named "WCFService*Test*". This script will create a clone of WCF GitHub repository, an IIS application pool and an IIS application from `src\System.Private.ServiceModel\tools\IISHostedWcfService`. The script will also install necessary test certificates and add IIS HTTPS binding for testing.
+
+This is one time setup. Once you set it up, you do not need to build or start the service manually. IIS will build and activate the service automatically upon the first request from tests. 
+
+Runing WCF Scenario Tests
+-------------------------
+As instructed in [Developer Guide](developer-guide.md#running-scenario-tests), to run scenario tests, you will need to specify the WCF service base address via `/p:ServiceUri` parameter to the build script.
```
Comment (roncain) [inline Documentation/scenario-test-guide.md]: My preference would be to make this scenario test guide *not* refer back to the developer guide but to keep all information about running scenario tests just in this scenario guide, and then to link to it from the developer guide.  I don't feel strongly either way so you choose which seems better.

**Comment by roncain** [comment]
LGTM.
I made some optional comments for consideration, but none of them are blocking.

File: Documentation/developer-guide.md [hunk 7]
```diff
@@ -23,58 +23,61 @@ Please read [Contributing](https://github.com/dotnet/wcf/blob/master/Documentati
 Building the repository
 =======================
 
-The WCF repository can be built from a regular, non-admin command prompt using build.cmd. This build produces a single System.Private.ServiceModel assembly that implements the individual client libraries.
+The WCF repository can be built from a regular, non-admin command prompt using build.cmd on Windows or build.sh on *nix.
```
Comment (iamjasonp) [inline Documentation/developer-guide.md]: NIT: consider using "non-Windows" or "other" platforms, not sure if people would associate OS X as "*nix"

File: Documentation/developer-guide.md (hunk 1 above)
Comment (zhenlan) [inline Documentation/developer-guide.md]: Fixed. thanks!

File: Documentation/developer-guide.md [hunk 8]
```diff
@@ -23,58 +23,61 @@ Please read [Contributing](https://github.com/dotnet/wcf/blob/master/Documentati
...
+```
 
-Running the scenario tests
+Running scenario tests
 ==========================
-Scenario tests are those tests that involve network activity between the tests and one or
-more running WCF services.  By default, they are not run with the normal build.cmd.
-To run the scenario tests, execute this CMD from the root of the repository:
+Scenario tests are those tests that involve network activity between the client tests and WCF test services.
+By default, they are not run with the normal build.cmd or build.sh.
```
Comment (iamjasonp) [inline Documentation/developer-guide.md]: NIT: remove "the" in "the normal build.cmd..."

File: Documentation/scenario-test-guide.md (hunk 3 above)
Comment (zhenlan) [inline Documentation/scenario-test-guide.md]: Fixed. Thanks!

File: Documentation/developer-guide.md [hunk 9]
```diff
@@ -23,58 +23,61 @@ Please read [Contributing](https://github.com/dotnet/wcf/blob/master/Documentati
...
-```
-
-Running scenario tests across multiple machines
-===============================================
-By default scenario tests run against WCF services that are locally self-hosted.
-To run these tests against WCF services running on a separate machine, see:
-[Cross-machine test guide](https://github.com/dotnet/wcf/blob/master/Documentation/cross-machine-test-guide.md).
-
-
-
```
Comment (iamjasonp) [inline Documentation/developer-guide.md]: Consider: the pitfall where if you change server, might want to point out deleting "Infrastructure.Common" might help

File: Documentation/scenario-test-guide.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+Scenario Testing Guide
+======================
+
+Overview
+--------
+WCF scenario tests are also referred to as outerloop tests. These are tests involving communication between WCF client tests and WCF test services. The WCF client tests running on .NET Core support cross platform, which means they can run on all supported platforms such as Windows, various Linux distros, Mac OS X as well as Universal Windows Platform (UWP). The WCF test services are built on full .NET Framwork that is supported on Windows only.
```
Comment (iamjasonp) [inline Documentation/scenario-test-guide.md]: NIT: "Framwork" should be "Framework"

File: Documentation/developer-guide.md (hunk 4 above)
Comment (zhenlan) [inline Documentation/developer-guide.md]: Sounds good. Added. Thanks!

File: Documentation/scenario-test-guide.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+```
+    StartWCFSelfHostedSvc.cmd
+```
+This script will add necessary firewall rules, generate and install self-signed test certificates needed by tests, build and kick off a self-hosted WCF test service process from project at `src\System.Private.ServiceModel\tools\SelfHostedWcfService`. When you are done with this WCF test service, type `exit` and press `enter` in the window where the service was started. Upon shutting down the service, this script will also clean up firewall rules and remove test certificates it installed.
+
+This script requires elevated privileges. If it is run from a non-elevated window, you may be prompted for elevated permission.
+
+IIS-Hosted WCF Test Service
+---------------------------
+To host WCF services on IIS, the Windows machine has these prerequisites
```
Comment (iamjasonp) [inline Documentation/scenario-test-guide.md]: Should we call out specific Windows versions?

File: Documentation/scenario-test-guide.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+```
+    ./build.sh /p:WithCategories=OuterLoop /p:ServiceUri=myserver
+```
+
+The `ServiceUri` for IIS hosted WCF service will be `{MachineName}/WCFService{id}`. For example, if you set up an IIS hosted WCF service with id *Test* on a Windows machine named *myserver*, you will be able to run scenario tests on a Linux or a Mac machine using this command
+
+```
+    ./build.sh /p:WithCategories=OuterLoop /p:ServiceUri=myserver/WcfServiceTest
+```
+To run scenario tests from a Windows machine, simply replace `./build.sh` with `build.cmd` in above commands.
```
Comment (iamjasonp) [inline Documentation/scenario-test-guide.md]: Style NIT: I don't like the word "simply" in documentation, typically... One person's simple is another person's horror  :smile:

**Comment by iamjasonp** [comment]
LGTM - no blocking comments, though it'd be good to have the spelling typos corrected at least

File: Documentation/scenario-test-guide.md (hunk 5 above)
Comment (zhenlan) [inline Documentation/scenario-test-guide.md]: Updated. Thanks!

File: Documentation/developer-guide.md (hunk 7 above)
Comment (zhenlan) [inline Documentation/developer-guide.md]: Agreed. Updated to "other platforms".

File: Documentation/developer-guide.md (hunk 8 above)
Comment (zhenlan) [inline Documentation/developer-guide.md]: Updated. Thanks!

File: Documentation/scenario-test-guide.md (hunk 12 above)
Comment (zhenlan) [inline Documentation/scenario-test-guide.md]: Agreed. Removed "simply" 😄. Thanks!

File: Documentation/scenario-test-guide.md (hunk 10 above)
Comment (zhenlan) [inline Documentation/scenario-test-guide.md]: Fixed. Thanks!

File: Documentation/developer-guide.md (hunk 9 above)
Comment (zhenlan) [inline Documentation/developer-guide.md]: Updated.

File: Documentation/scenario-test-guide.md (hunk 6 above)
Comment (zhenlan) [inline Documentation/scenario-test-guide.md]: Yes, that's probably a better way. Didn't get time to update it this time. Next time 😄.

**Pull request merged**